### PR TITLE
MM-40841, MM-40916: Templates Selector, FTUE flow

### PR DESCRIPTION
**Pull request by calebroseland**
#### Summary
- Updated Template Selector, now more friendly looking and below playbooks list
- FTUE content and minor UX flow improvements
- Template changes:
  - ~Service Reliability Incident~ -> Incident Resolution

(Before)
CleanShot 2022-02-01 at 09 22 42@2x

(After)
CleanShot 2022-02-03 at 11 04 03@2x

(Responsive; selector when there're runs but no playbooks)
CleanShot 2022-01-31 at 17 07 10

(Start; when no runs and no playbooks, redirects to `/playbooks/start` and nav bar disappears)
CleanShot 2022-02-03 at 11 12 56@2x

#### Tech. notes

- For clarity and easier reading, `template_selector.tsx` has been split/refined into:
  -> `template_data.tsx`
  -> `template_selector.tsx`
  -> `template_item.tsx`
- Default/root route now properly redirects to `/playbooks/runs`
this avoids the issue where the Runs nav item was not active (it's now blue on default route)
- FTUE routing takes place in `runs_page`
  - If there are `no runs`, and `no playbooks`—FTUE will kick in and redirect to `/playbooks/start`, this is just `playbook_list` with a special route to engage FTUE and hide nav bar)
  - If there are runs, it will stay on the runs page to prevent losing access to runs of archived playbooks or where odd situations/migrations may have happened

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-40841
- https://mattermost.atlassian.net/browse/MM-40916

#### Checklist

Item | Y / N / NA [^1]
---|---
Telemetry updated? | 
Gated by experimental feature flag? | N
Unit/E2E tests updated? | 

[^1]: Yes / No / Not Applicable.

### CODE REVIEW

File: webapp/src/components/backstage/playbook_list.tsx [hunk 1]
```diff
@@ -36,37 +37,85 @@ import {
...
+    line-height: 28px;
+    text-align: center;
+`;
+
+const AltSub = styled(Sub)`
+    text-align: center;
+    margin-bottom: 36px;
+`;
+
+const PlaybookList = (props: {ftue?: boolean}) => {
```
Comment (crspeller) [inline webapp/src/components/backstage/playbook_list.tsx:114]: Confusing if I don't know what `ftue` stands for. Maybe do `firstTimeUser`?

File: webapp/src/components/templates/template_selector.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            isClearable={false}
+            maxMenuHeight={380}
+        />
+    );
+};
+
+const SelectorGrid = styled.div`
+	display: grid;
+	grid-template-columns: repeat(auto-fill, minmax(260px, 1fr));
+	/* grid-template-rows: repeat(auto-fill, minmax(256px, 1fr)); */
```
Comment (crspeller) [inline webapp/src/components/templates/template_selector.tsx:63]: Nit: remove

File: webapp/src/components/templates/template_selector.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import TemplateItem from './template_item';
+import PresetTemplates, {PresetTemplate} from './template_data';
+
+const presetTemplateOptions = PresetTemplates.map((template: PresetTemplate) => ({label: template.title, value: template.title}));
+
+interface Props {
+    templates?: PresetTemplate[];
+}
+
+export function isPlaybookCreationAllowed(allowPlaybookCreationInTeams: Map<string, boolean>) {
```
Comment (crspeller) [inline webapp/src/components/templates/template_selector.tsx:25]: Probably doesn't belong here.

**Review by crspeller — APPROVED** [review]
Nice! Just a couple nits.

File: webapp/src/components/assets/illustrations/dumpster_fire_svg.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+const DumpsterFire = (props: {className?: string}) => (
+    <Svg
+        width='60'
+        height='72'
+        viewBox='0 0 60 72'
+        fill='none'
+    >
+        <path
+            d='M7.78885 63.3345C9.29727 63.3345 10.5201 62.1083 10.5201 60.5957C10.5201 59.0831 9.29727 57.8569 7.78885 57.8569C6.28043 57.8569 5.05762 59.0831 5.05762 60.5957C5.05762 62.1083 6.28043 63.3345 7.78885 63.3345Z'
+            fill='#1F5C43'
```
Comment (iomodo) [inline webapp/src/components/assets/illustrations/dumpster_fire_svg.tsx:18]: Should these colours be theme independent?

File: webapp/src/components/backstage/runs_page.tsx [hunk 5]
```diff
@@ -37,27 +41,31 @@ const RunListContainer = styled.div`
...
 
         checkForPlaybookRuns();
     }, []);
 
-    if (showNoPlaybookRuns) {
+    if (showNoPlaybookRuns == null || noPlaybooks == null) {
+        return null;
+    }
+
+    if (showNoPlaybookRuns === true) {
```
Comment (iomodo) [inline webapp/src/components/backstage/runs_page.tsx:65]: Do we need `true` here?

File: webapp/src/components/templates/template_item.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+const Item = styled.div`
+    position: relative;
+    cursor: pointer;
+    border: 1px solid rgba(var(--center-channel-color-rgb), 0.08);
+    background: var(--center-channel-bg);
+    border-radius: 8px;
+    box-sizing: border-box;
+    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.08);
+    aspect-ratio: 284 / 320;
```
Comment (abhijit-singh) [inline webapp/src/components/templates/template_item.tsx:29]: We can make this `aspect-ratio: 284 / 300;`
The cards seem too tall right now with the higher width.

**Review by abhijit-singh — CHANGES_REQUESTED** [review]
Looks great @calebroseland! Added some minor comments and a couple more requests:
1. Can we add a hover state to the cards like in the designs:
image
2. For the `Recommended for beginners` tag, can we hardcode the text color to `#CC8F00`. It should work in light and dark themes based on my tests in Figma. If not, we can leave it as is.

File: webapp/src/components/backstage/runs_page.tsx (hunk 5 above)
Comment (calebroseland) [reply webapp/src/components/backstage/runs_page.tsx:65]: Not anymore (did when the return null was after these redirect checks)

**Comment by calebroseland** [comment]
re: 1. Will add hover state ✓.
re: 2. I can, but it in dark themes it feels a bit too low-contrast (ratio: 3.12)
CleanShot 2022-02-08 at 16 24 26@2x

File: webapp/src/components/assets/illustrations/dumpster_fire_svg.tsx (hunk 4 above)
Comment (calebroseland) [reply webapp/src/components/assets/illustrations/dumpster_fire_svg.tsx:18]: I didn't see any themed colors specifically re: these template illustrations, but I did make the rocket_man shadow themed which wasn't showing earlier. Thanks for mentioning!
CleanShot 2022-02-09 at 18 17 45@2x

**Comment by calebroseland** [comment]
@abhijit-singh cc @iomodo to keep things moving I'm going to go ahead and merge now after the latest revisions—thanks for the callouts! Any follow up changes/iterations can be done as-needed.

**Pull request merged**